apps/studio/src/components/PageEditor/ComponentSelector.tsx

```diff
@@ -103,7 +104,17 @@ function ComponentSelector() {
     setDrawerState,
     setSavedPageState,
     setPreviewPageState,
+    setAddedBlock,
   } = useEditorDrawerContext()
+
+  // TODO: get this dynamically
+  const pageId = 1
+  const siteId = 1
```

Comment:
i think we can now pass in these as props for this component from the page that uses this (or put it into `EditorDrawerContext`)

Reply:
updated!

i think the drawer context shouldn't be so bloated. once the 2 PRs from me go in, i'll write a PR to refactor the callsites of `useQueryParse` now so that it'll be done via a context